doc/source/ray-core/tasks/generators.rst

```diff
@@ -0,0 +1,78 @@
+.. _generators:
+
+Generators
+==========
+
+Python generators are functions that behave like an iterator, yielding one
+value per iteration. Ray supports remote generators for two use cases:
+
+1. To reduce max heap memory usage when returning multiple values from a remote
+   function. See the :ref:`design pattern guide <generator-pattern>` for an
+   example.
+2. When the number of return values is set dynamically by the remote function
+   instead of by the caller.
+
+
+`num_returns` set by the task caller
+------------------------------------
+
+Where possible, the caller should set the remote function's number of return values using ``@ray.remote(num_returns=x)`` or ``foo.options(num_returns=x).remote()``.
```

Comment:
Could we explain why: why static generator is preferred where possible?

Reply:
Yes, good point.